### DISCUSSION TITLE
How to keep a persistent volume with AddSqlServerContainer?

### DISCUSSION THREAD

**Discussion by julioct** [microsoft/aspire#1109] [Q&A] [answered]
I am able to use AddSqlContainer successfully in my AppHost project. The problem is that every time I stop AppHost, the container is destroyed (which is fine) but also all the data I had in the DB is destroyed with it.

I'd like to have a way to specify a persistent volume that should survive beyond the lifetime of the AppHost and get reconnected on subsequent runs.

For reference, this is my current setup:

```csharp
var builder = DistributedApplication.CreateBuilder(args);

var pass = builder.Configuration["SqlPassword"];
var sql = builder.AddSqlServerContainer("sql", pass, 1433)
                 .AddDatabase("GameStore");

var backend = builder.AddProject<Projects.GameStore_Api>("backend")
                     .WithReference(sql);

builder.AddProject<Projects.GameStore_Frontend>("frontend")
       .WithReference(backend);

var app = builder.Build();

app.Run();
```

And this is how I run my SQL Server container manually today:
```
docker run -e "ACCEPT_EULA=Y" -e "MSSQL_SA_PASSWORD=$sa_password" -p 1433:1433 -v sqlvolume:/var/opt/mssql -d --rm --name mssql mcr.microsoft.com/mssql/server:2022-latest
```

Also, I already tried using **WithVolumeMount()** but it makes no difference, the data is gone.

**Comment by DamianEdwards** [comment]
Related #923 #972

**Comment by recepgunes1** [comment]
I had similar issue with you. When I change VolumeMountType from Bind to Named,  it solved. I think this is going to solve your problem. At least it worked for elasticsearch.

Also, default VolumeMountType is Bind.

```
        builder.AddSqlServerContainer("sql", "Passw123.", 1433)
            .AddDatabase("GameStore")
            .WithVolumeMount("sqlvolume", "/var/opt/mssql", VolumeMountType.Named);

```

**Reply by trackjack-dev** [reply]
when i add .WithVolumeMount() and I look at the docker run command of the container, there is no -v option added. It also doesn't create the volume. Is this a bug or am I the only one who is left with this issue?

**Comment by davidfowl** [comment, marked as answer]
There's an example with volume mounts here https://github.com/dotnet/aspire-samples/tree/main/samples/DatabaseContainers

**Reply by julioct** [reply]
Beautiful, thank you!

**Reply by trackjack-dev** [reply]
Thank you. There is a lot of improvements and great new stuff in preview 3!
